base_menu_visibility_restriction/models/ir_ui_menu.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models, tools
+
+
+class IrUiMenu(models.Model):
+
+    _inherit = "ir.ui.menu"
+
+    excluded_groups_id = fields.Many2many(
+        comodel_name="res.groups",
+        relation="ir_ui_menu_excluded_group_rel",
+        column1="menu_id",
+        column2="gid",
+        string="Excluded Groups",
+    )
+
+    @api.model
+    @tools.ormcache("frozenset(self.env.user.groups_id.ids)", "debug")
+    def _visible_menu_ids(self, debug=False):
+        """ Return the ids of the menu items visible to the user. """
+        # retrieve all menus, and determine which ones are visible
```

Comment:
@victoralmau I have just noticed you are rewritting this method without calling the super method. That can break some functionallity. Is it really necessary?

Reply:
Super improved and simplify function.